executor/executor.go

```diff
@@ -320,8 +334,17 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 
 		fReader := bufio.NewReader(f)
 
+		exportEnvCommand := []string{
+			"set -e",
+			"export PATH=$PATH:/opt/sd",
+			exportEnvCmd,
+		}
+
 		go func() {
 			runCode, rcErr := doRunCommand(guid, stepFilePath, emitter, f, fReader)
+			// run export file after step execution
+			shargs := strings.Join(exportEnvCommand, " && ")
```

Comment:
env variable created as part of step is not available by default after each step?

Reply:
yeah, it's not, it gets exported as part of the trap that is run after step exit. However, in case of abort that does not happen and so we have to save the export. for normal builds it should be overwritten by the export in trap function.

Reply:
it's actually available as part of each step since it's the same shell. 
What we are trying to do is when a new shell i started  (due to step failure or abort or timeout) env variables are exported to it.

Reply:
ok, so existing behavior is teardown steps will not receive any env variable that's set as part of user steps.

Reply:
not exactly, when shell terminates due to step failure, existing `trap` logic works to export env, this new flow is needed only for abort flow, where the `trap` is not getting triggered.

Reply:
to clarify when we abort a build, do sd teardown steps receive the user set environment variables? my assumption is no.